core/platform/search/gae_search_services_test.py

```diff
@@ -584,6 +599,21 @@ def test_search_using_single_sort_expression(self):
         self.assertEqual(result[1].get('id'), 'doc1')
         self.assertEqual(result[2].get('id'), 'doc2')
 
+    def test_raise_error_when_sort_starts_with_invalid_character(self):
+        doc = {'id': 'doc1', 'k': 'abc def', 'rank': 3, 'language_code': 'en'}
+        gae_search_services.add_documents_to_index([doc], 'index')
+        # Fields in the sort expression need to start with '+' or '-'
+        # to indicate sort direction. If no such indicator is there, it will
+        # raise ValueError.
+        sort_expression = 'k'
```

Comment:
You can inline this, and perhaps use something plausible (that connotes sorting) but that's not "k", to distinguish from k:abc.

Reply:
Done